bin/rundownloadgithub.sh

```diff
@@ -12,99 +12,61 @@
-fi
-cd $NONIETFDIR/openconfig/public
-git pull --recurse-submodules https://github.com/openconfig/public.git >>$LOG 2>&1
-
-# Trick below is "flatten" all .yang files from subdirectories into one directory, to avoid the ConfD path issues
-cd $NONIETFDIR/openconfig/public/release/models
-if [ -d $NONIETFDIR/openconfig-flat/public/release/models ]; then
-	rm -rf $NONIETFDIR/openconfig-flat/public/release/models
-fi
-mkdir -p $NONIETFDIR/openconfig-flat/public/release/models
-find . -name "*.yang" -exec cp -t $NONIETFDIR/openconfig-flat/public/release/models/ {} + >>$LOG 2>&1
+refresh_repo () {
+    local repo=$1
+    local owner=${1%%/*}
+    local directory=${1##*/}
+    if [ ! -d $NONIETFDIR/$repo ]
+    then
+        mkdir -p $NONIETFDIR/$owner
+        cd $NONIETFDIR/$owner
+        git clone --recurse-submodules ${REPOSITORIES[$repo]} $directory >> $LOG 2>&1
+    else
+        cd $NONIETFDIR/$repo
+        if [ "$(git pull origin HEAD 2>&1 | tee -a $LOG)" != "Already up to date." ]
+        then
+            modified+=([$repo]=true)
+        fi
+        git submodule update --init --recursive >> $LOG 2>&1
+    fi
+}
 
-# get the entire content of github (sysrepo)
-mkdir -p $NONIETFDIR/sysrepo
-if [ ! -d $NONIETFDIR/sysrepo/yang ]; then
-	cd $NONIETFDIR/sysrepo
-	git clone --recurse-submodules https://github.com/sysrepo/yang.git >>$LOG 2>&1
-fi
-cd $NONIETFDIR/sysrepo/yang
-git init >>$LOG 2>&1
-git pull --recurse-submodules https://github.com/sysrepo/yang.git >>$LOG 2>&1
+if [ "$1" != "--test" ]
+then
+    # Get the local configuration
+    source configure.sh
+    LOG=$LOGS/downloadGitHub.log
+    date +"%c: Starting" >$LOG
 
-# get the entire content of github (ONF)
-mkdir -p $NONIETFDIR/onf
-if [ ! -d $NONIETFDIR/onf/Snowmass-ONFOpenTransport ]; then
-	cd $NONIETFDIR/onf
-	git clone --recurse-submodules https://github.com/OpenNetworkingFoundation/Snowmass-ONFOpenTransport.git >>$LOG 2>&1
-fi
-cd $NONIETFDIR/onf/Snowmass-ONFOpenTransport
-git init >>$LOG 2>&1
-git pull --recurse-submodules https://github.com/OpenNetworkingFoundation/Snowmass-ONFOpenTransport.git >>$LOG 2>&1
+    for repo in "${!REPOSITORIES[@]}"
+    do
+        refresh_repo $repo
+    done
 
-# get the entire content of OpenROADM (ONF)
-mkdir -p $NONIETFDIR/openroadm
-if [ ! -d $NONIETFDIR/openroadm/OpenROADM_MSA_Public ]; then
-	cd $NONIETFDIR/openroadm
-	git clone --recurse-submodules https://github.com/OpenROADM/OpenROADM_MSA_Public.git >>$LOG 2>&1
-fi
-cd $NONIETFDIR/openroadm/OpenROADM_MSA_Public
-git init >>$LOG 2>&1
-git pull --recurse-submodules https://github.com/OpenROADM/OpenROADM_MSA_Public.git >>$LOG 2>&1
+    # "Flatten" all .yang files from subdirectories into one directory to avoid confd path issues.
+    if [ ${modified[openconfig/public]} ]
+    then
+        cd $NONIETFDIR/openconfig
+        rm -rf public-flat
+        cp public public-flat
+        cd public-flat
+        mkdir -p release/models-flat
```

Comment:
Following symbolic link is created in `runIETFstats.sh` script:
`ln -f -s $NONIETFDIR/openconfig-flat/public/release/models/ $MODULES/openconfig-main`
So directory name must be changed either here or in the `runIETFstats.sh` script:

Reply:
that's true, but symbolic link refer to the `$NONIETFDIR/openconfig-flat/public/release/models/` and you are copying files into `$NONIETFDIR/openconfig/public-flat/release/models/`
This inconsistency needs to be resolved